### PR TITLE
Use `Dir3` in `Transform` APIs

### LINKED ISSUE #12481: Make more `Transform` APIs interoperate with `Dir3`
## What problem does this solve or what need does it fill?

Presently, `Transform::look_to` and `Transform::align` (along with other APIs derived from them) take `Vec3` inputs even when their parameters are, morally speaking, directions. Conversions between `Vec3` and `Dir3` are annoying for end-users, and some of our other APIs provide `Dir3`, while doing mathematics 'by hand' often results in a `Vec3`, so simultaneously capturing these would be nice.

## What solution would you like?

After some discussion on Discord, it appears that using `impl TryInto<Dir3>` would be natural for these cases, since they
1. treat their input essentially as a direction even though it's a vector
2. already provide replacement input in the case that the conversion `Vec3 -> Dir3` would fail

Thus, in particular, changing these inputs from `Vec3` to `impl TryInto<Dir3>` is not even a breaking change, but it would allow `Dir3` input to be given just as easily as `Vec3`.

## What alternative(s) have you considered?

Users can manually convert between `Vec3` and `Dir3` to get around this.

## Additional context

On the level of API design, just using `Dir3` was considered for `Transform::align` because it is more "natural", but providing a `Dir3` is often not ergonomic (especially since the `Vec3` conversion is fallible) and because `Transform::look_to` used `Vec3` already.

### PR DESCRIPTION
**Pull request by yyogo**
# Objective

Make `Transform` APIs more ergonomic by allowing users to pass `Dir3` as an argument where a direction is needed. Fixes #12481.

## Solution

Accept `impl TryInto<Dir3>` instead of `Vec3` for direction/axis arguments in `Transform` APIs

---

## Changelog
The following `Transform` methods now accept an `impl TryInto<Dir3>` argument where they previously accepted directions as `Vec3`:
* `Transform::{look_to,looking_to}`
* `Transform::{look_at,looking_at}`
* `Transform::{align,aligned_by}`

## Migration Guide

This is not a breaking change since the arguments were previously `Vec3` which already implements `TryInto<Dir3>`, and behavior is unchanged.

### CODE REVIEW

**Review by afonsolage — COMMENTED** [review]
I wonder if shouldn't we use `Dir3` directly, instead of `Vec3`, since `Dir3` makes it explicit what is going on and what to expect from those parameters.

For the sake of avoiding breaking changes, it could make sense to use `impl TryInto<Dir3>`, but if we use `Dir3` we don't have to worry about fallback values, since `Dir3` will always be normalized and always represents an unit direction.

**Comment by yyogo** [comment]
@afonsolage This is mentioned in the linked issue:

> On the level of API design, just using Dir3 was considered for Transform::align because it is more "natural", but providing a Dir3 is often not ergonomic (especially since the Vec3 conversion is fallible) and because Transform::look_to used Vec3 already.

I'm fairly neutral on this tbh, I implemented the issue as specified but both options make sense to me.

**Review by djeedai — APPROVED** [review]

**Review by afonsolage — APPROVED** [review]
Ok, in this case we are going for `impl TryInto<Dir3>` instead of `Dir3` due to ergonomics and DX.

File: crates/bevy_transform/src/components/transform.rs [hunk 1]
```diff
@@ -122,11 +122,11 @@ impl Transform {
     ///
     /// In some cases it's not possible to construct a rotation. Another axis will be picked in those cases:
     /// * if `target` is the same as the transform translation, `Vec3::Z` is used instead
-    /// * if `up` is zero, `Vec3::Y` is used instead
+    /// * if `up` fails converting to `Dir3`, `Dir3::Y` is used instead
```
Comment (IQuick143) [inline crates/bevy_transform/src/components/transform.rs:125]:
```suggestion
    /// * if `up` fails converting to `Dir3` (e.g if it is `Vec3::ZERO`), `Dir3::Y` is used instead
```

File: crates/bevy_transform/src/components/transform.rs [hunk 2]
```diff
@@ -373,100 +373,100 @@ impl Transform {
...
-    /// t1.align(Vec3::ZERO, Vec3::Z, Vec3::ZERO, Vec3::X);
-    /// t2.align(Vec3::X, Vec3::Z, Vec3::Y, Vec3::X);
+    /// t1.align(Vec3::ZERO, Dir3::Z, Vec3::ZERO, Dir3::X);
+    /// t2.align(Dir3::X, Dir3::Z, Dir3::Y, Dir3::X);
     /// assert_eq!(t1.rotation, t2.rotation);
     ///
-    /// t1.align(Vec3::X, Vec3::Z, Vec3::X, Vec3::Y);
+    /// t1.align(Dir3::X, Dir3::Z, Dir3::X, Dir3::Y);
     /// assert_eq!(t1.rotation, Quat::from_rotation_arc(Vec3::X, Vec3::Z));
     /// ```
```
Comment (IQuick143) [inline crates/bevy_transform/src/components/transform.rs:443]: Would be nice if the other methods also gotten an example like this, showcasing how you can mix&match vectors and directions to improve discoverability of the fact that both Dir3 and Vec3 are valid arguments, for new users.

Not a necessary change though.

**Review by IQuick143 — APPROVED** [review]

**Comment by NthTensor** [comment]
TryInto seems like the right choice to me, but I haven't looked this over super closely.

Nice work!

**Pull request merged**